File: backend/src/main/java/be/lutske/leolegacy/infrastructure/persistence/repository/RecipeRepository.java
package be.lutske.leolegacy.infrastructure.persistence.repository;

import be.lutske.leolegacy.infrastructure.persistence.entity.RecipeEntity;
import io.quarkus.hibernate.orm.panache.PanacheRepository;
import jakarta.enterprise.context.ApplicationScoped;

import java.util.List;

@ApplicationScoped
public class RecipeRepository implements PanacheRepository<RecipeEntity> {

    public List<RecipeEntity> findByCategoryId(long categoryId) {
        return list("category.id", categoryId);
    }

    public List<RecipeEntity> findTopByViewCount(int limit) {
        return list("ORDER BY viewCount DESC").stream()
                .limit(limit)
                .toList();
    }
}

File: backend/src/main/kotlin/be/lutske/leolegacy/infrastructure/persistence/repository/RecipeRepository.kt
package be.lutske.leolegacy.infrastructure.persistence.repository

import be.lutske.leolegacy.infrastructure.persistence.entity.RecipeEntity
import io.quarkus.hibernate.orm.panache.kotlin.PanacheRepository
import jakarta.enterprise.context.ApplicationScoped

@ApplicationScoped
class RecipeRepository : PanacheRepository<RecipeEntity> {

    fun findByCategoryId(categoryId: Long): List<RecipeEntity> =
        list("category.id", categoryId)

    fun findTopByViewCount(limit: Int): List<RecipeEntity> =
        list("ORDER BY viewCount DESC", *arrayOf()).stream().limit(limit.toLong()).toList()
}
